fbpic/lpa_utils/laser/longitudinal_laser_profiles.py

```diff
@@ -229,40 +233,71 @@ def __init__(self, z0, spectrum_file, propagation_direction=1):
             in the file. The spectral phase can be omitted, in which case it
             will be assumed to be 0.
 
+        phi2_chirp: float (in second^2), optional
+            The amount of temporal chirp, at focus (in the lab frame).
+
+        phi3_chirp: float (in second^3), optional
+            The amount of third order dispersion, at focus (in the lab frame).
+            This parameter leads to an asymmetrical temporal shape of the laser
+            pulse.
+
+        phi4_chirp: float (in second^4), optional
+            Fourth order dispersion at focus (in the lab frame).
+
+        subtract_linear_phase: bool, optional
+            If True, a linear slope in the spectral phase read
+            from spectrum_file will be subtracted from it. A non-zero
+            linear slope leads to pulse being shifted in time.
+
         propagation_direction: int, optional
             Indicates in which direction the laser propagates.
             This should be either 1 (laser propagates towards positive z)
             or -1 (laser propagates towards negative z).
         """
         # Initialize propagation direction
-        LaserLongitudinalProfile.__init__(self,propagation_direction,
+        LaserLongitudinalProfile.__init__(self, propagation_direction,
                                           gpu_capable=False)
 
         # Load the spectrum from the text file
         spectrum_data = np.loadtxt( spectrum_file, delimiter='\t' )
-        wavelength = spectrum_data[:,0]
-        intensity = spectrum_data[:,1]
+        wavelength = spectrum_data[:, 0]
+        intensity = spectrum_data[:, 1]
         if spectrum_data.shape[1] < 3:
-            phase = np.zeros_like(wavelength)
+            spectral_phase = np.zeros_like(wavelength)
         else:
-            phase = spectrum_data[:,2]
+            spectral_phase = spectrum_data[:, 2]
+        omega_axis = 2 * np.pi * c / wavelength
+
+        # Subtract linear slope, if asked to
+        if subtract_linear_phase:
+            # Fit a 4 order polynomial and subtract the linear term
+            poly_coeffs = np.polyfit(omega_axis, spectral_phase, 4)
+            spectral_phase -= omega_axis * poly_coeffs[-1]
```

Comment:
```suggestion
            spectral_phase -= omega_axis * poly_coeffs[-2]
```
My understanding of the `numpy.polyfit` interface is that `poly_coeffs[-1]` is the constant term of the polynomial, while `poly_coeffs[-2]` is the linear term. Do you agree?